streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractRocksDBSegmentedBytesStore.java

```diff
@@ -87,23 +90,35 @@ public KeyValueIterator<Bytes, byte[]> backwardFetch(final Bytes key,
         return fetch(key, from, to, false);
     }
 
+
     KeyValueIterator<Bytes, byte[]> fetch(final Bytes key,
                                           final long from,
                                           final long to,
                                           final boolean forward) {
-        final List<S> searchSpace = keySchema.segmentsToSearch(segments, from, to, forward);
 
-        final Bytes binaryFrom = keySchema.lowerRangeFixedSize(key, from);
+        final long actualFrom = getActualFrom(from);
+
+        if (keySchema instanceof WindowKeySchema && to < actualFrom) {
+            return KeyValueIterators.emptyIterator();
```

Comment:
Could you help me understand why we only return `emptyIterator` when it's `WindowKeySchema`? Why not other types of schema?

Reply:
Yeah.. so for the case of WindowKeySchema, if to < from, the underlying code throws IllegalArguementException while that's not the case for SessionKeySchema and other kinds of schema.